addon/components/flexberry-layerstree.js

```diff
@@ -0,0 +1,167 @@
+          name="hbsLayer1.2 (it is a leaf)"
+          type="wms"
+          visibility=true
+        }}
+      {{/flexberry-layerstree}}
+    {{/flexberry-layerstreenode}}
+  {{/flexberry-layerstree}}
+  ```
+
+  templates/my-form-in-json.hbs
+  ```handlebars
+  {{flexberry-layerstree
+    layers=(get-with-dynamic-actions this "jsonLayersTreeNodes"
+      hierarchyPropertyName="layers"
+      dynamicActions=(array
+        (hash
+          on="visibilityChange"
+          actionName="onTreenodeVisibilityChange"
+          actionArguments=(array "{% propertyPath %}.visibility")
+        )
+        (hash
+          on="headerClick"
+          actionName="onTreenodeHeaderClick"
+          actionArguments=(array "{% propertyPath %}")
+        )
+      )
+    )
+  }}
+  ```
+
+  @class FlexberryLayersTreeComponent
+  @extends FlexberryTreeComponent
+*/
+let FlexberryLayersTreeComponent = FlexberryTreeComponent.extend({
+  /**
+    Flag: indicates whether some {{#crossLink "FlexberryLayersTreeComponent/layers:property"}}tree 'layers'{{/crossLink}} are defined.
+
+    @property _hasLayers
+    @type boolean
+    @readonly
+    @private
+  */
+  _hasLayers: Ember.computed('layers.[]', function() {
+    let layers = this.get('layers');
+
+    return Ember.isArray(layers) && layers.length > 0;
+  }),
+
+  /**
+    Flag: indicates whether tree isn't placed inside {{#crossLink "FlexberryLayersTreeComponent"}}flexberry-layerstreenode component{{/crossLink}}.
+
+    @property _isNotInsideTreeNode
+    @type Boolean
+    @readonly
+    @private
+  */
+  _isNotInsideTreeNode: Ember.computed('parentView', function() {
+    let parentView = this.get('parentView');
+
+    return !(parentView instanceof FlexberryLayersTreenodeComponent);
+  }),
+
+  /**
+    Reference to component's template.
+  */
+  layout,
+
+  /**
+    Reference to component's CSS-classes names.
+    Must be also a component's instance property to be available from component's .hbs template.
+  */
+  flexberryClassNames,
+
+  /**
+    Component's wrapping <div> CSS-classes names
+    (extra to the {{#crossLink "FlexberryTreeComponent/classNames:property"}}base ones{{/crossLink}}).
+
+    Any other CSS-class names can be added through component's 'class' property.
+    ```handlebars
+    {{#flexberry-layerstree class="styled"}}
+      Tree's content
+    {{/flexberry-layerstree}}
+    ```
+
+    @property classNames
+    @type String[]
+    @default ['flexberry-layerstree', 'flexberry-tree', 'accordion']
+  */
+  classNames: [flexberryClassNames.wrapper],
+
+  /**
+    Map's layers.
+    This property is optional and must be used when there are too many child layers,
+    and you don't want to define them explicitly in the .hbs markup,
+    then you can define layers array somewhere in code & pass defined array to this component's property.
+
+    @property layers
+    @type NewPlatformFlexberryGISMapLayerModel[]
+    @default undefined
+  */
+  layers: undefined

```

Comment:
To my mind it should be 'undefined'. I'll remove strokes like "@default undefined".